### DISCUSSION TITLE
Empty catch {} block causing issues with babel.

### DISCUSSION THREAD

**Discussion by keating99** [babel/babel#14934] [Q&A]
Hi all,

I'm just wondering if anyone else has come across the issue with babel and webpack where the ssh2 module doesn't parse because of catch {} littered throughout the code?

One example of this is:

Module parse failed: C:\Users\...node_modules\ssh2\lib\client.js Unexpected token (315:16)
[0] You may need an appropriate loader to handle this file type.
[0] |         try {
[0] |           sock.end();
[0] |         } catch {} << this is the offending line of code 
[0] |       },

Maybe there is a babel plugin that deals with this issue? I can't seem to work it out, I'd imagine this is a fairly common issue that I have yet to find a solution for.

I have tried @babel/plugin-proposal-optional-catch-binding but it doesn't seem to do what I need. This syntax have been supported since node v10.16.0 I'm currently running v12.

I'm sure that people have come across this issue in the past. 

Any help or pointers from anyone out there would be great. Thanks

**Comment by liuxingbaoyu** [comment]
https://github.com/babel/babel/issues/14829
This is probably not a problem with babel, it would be great if you could provide a reproducible repository.

**Comment by JLHwung** [comment]
> C:\Users...node_modules\ssh2\lib\client.js

The error is thrown from your node_modules. Did you exclude node_modules in your babel-loader config? If so please refer to https://github.com/babel/babel-loader#some-files-in-my-node_modules-are-not-transpiled-for-ie-11

**Reply by keating99** [reply]
Updating Webpack from 3 to 4 solved the problem.
